### PR TITLE
Server list branch

### PR DESCRIPTION
**Pull request by TchaikovDriver**
1. Add an menu item "Server list" in MainActivity, which makes it possible to cache several Trojan profiles and switch to one of them swiftly. Also add an SAVE button in MainActivity to save current Trojan configuration in server list cache file.

2. Add an menu item in ServerListActivity to scan QR code which contains TrojanURL. Once a TrojanURL is read, it will be parsed to TrojanConfig and save in server list cache file.

3. Add an menu item "Import from file" in ServerListActivity, offering a quick way to import serveral profiles. The file must be in JSON format, it is assumed that the file starts with an JSON Array whose key is "configs". In the "configs", each Trojan profile is assumed to contain properties below: "server"(remote address), "server_port"(remote port, type int), "password" and "verify".

### CODE REVIEW

**Comment by firogh** [comment]
Great job！Thank you.

**Comment by llcat** [comment]
good job! hope release quickly

File: app/src/main/java/io/github/trojan_gfw/igniter/MainActivity.java [hunk 1]
```diff
@@ -44,10 +51,59 @@
...
+        }
+    };
+    private TextViewListener passwordTextListener = new TextViewListener() {
+        @Override
+        protected void onTextChanged(String before, String old, String aNew, String after) {
+            // update TextView
+            startUpdates(); // to prevent infinite loop.
+            if (passwordText.hasFocus()) {
+                TrojanConfig ins = Globals.getTrojanConfigInstance();
+                Log.i(TAG, "set instance password text: " + passwordText.getText().toString());
```
Comment (oasiscifr) [inline app/src/main/java/io/github/trojan_gfw/igniter/MainActivity.java]: It seems for debug? IMO it is sensitive information.

Comment (TchaikovDriver) [reply app/src/main/java/io/github/trojan_gfw/igniter/MainActivity.java]: Oops, you're right. I will remove the log at the next commit.

File: app/src/main/java/io/github/trojan_gfw/igniter/servers/data/ServerListDataManager.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    @Override
+    public void deleteServerConfig(TrojanConfig config) {
+        List<TrojanConfig> trojanConfigs = loadServerConfigList();
+        for (int i = trojanConfigs.size() - 1; i >= 0; i--) {
+            if (trojanConfigs.get(i).getRemoteAddr().equals(config.getRemoteAddr())) {
+                trojanConfigs.remove(i);
+                break;
+            }
+        }
+
```
Comment (oasiscifr) [inline app/src/main/java/io/github/trojan_gfw/igniter/servers/data/ServerListDataManager.java]: Forget to call `replaceServerConfigs` ?

Comment (TchaikovDriver) [reply app/src/main/java/io/github/trojan_gfw/igniter/servers/data/ServerListDataManager.java]: Yes, my bad. Already fixed it.

**Comment by TchaikovDriver** [comment]
@oasiscifr Fix the problems you mentioned and the bug I found.

**Comment by oasiscifr** [comment]
https://github.com/trojan-gfw/igniter/issues/77 solved

**Pull request merged**